snowshu/core/configuration_parser.py

```diff
@@ -94,8 +102,49 @@ def _get_dict_from_anything(self, dict_like_object:Union[str,'StringIO',dict])->
             try:
                 return yaml.safe_load(dict_like_object.read())
             except AttributeError:
-                with open(dict_like_object,'r') as f:
-                    return yaml.safe_load(f.read())
+                return self._load_and_verify_file(dict_like_object, **kwargs)
+
+    def _load_and_verify_file(self,
+                              file_path,
+                              verify: bool = True,
+                              templates_path: Path = Path(os.path.dirname(__file__)).parent / 'templates',
+                              schema_path: Path = None):
+        logger.debug("Parsing file at %s", file_path)
+        if not schema_path:
+            schema_path = templates_path / f'{Path(file_path).stem}_schema.yml'
+        if not os.path.isfile(schema_path):
+            logger.warning("Loading from %s, but no schema (%s) found to verify it.",
+                            file_path, schema_path)
+            verify = False
+
+        # Load data
+        try:
+            with open(file_path, 'r') as f:
+                loader = AnsibleLoader(f)
+                instance = loader.get_single_data()
+        except FileNotFoundError:
+            loader = None
+            raise
+        except Exception as exc:
+            raise Exception(f'Error when loading YAML config at {file_path} {exc}')
+        finally:
+            if loader:
+                loader.dispose()
+
+        if verify:
```

Comment:
I'm pretty sure we only really need this block

Reply:
Slightly related. Is the `verify` flag meant for the case when we have a `yaml` config file that doesn't have a schema file specified for it?

If we are planning on using json schema for everything, I don't think that should ever be the case, and we could have that path error out entirely.

Reply:
Paths can error out entirely if we have a different file names for our input files, I do not think `replica.yml` can be renamed, but the `credpath` in `replica.yml` can contain any name and this can lead to `FileNotFoundError` when looking for schema for the bad credentials file.
https://github.com/Health-Union/snowshu/pull/54/files#diff-ec76ef37dcbb1075645b9ef7cb0021659dad0a01b09ea3908846911999b451f4R111